core/src/main/java/hudson/model/listeners/ItemListener.java

```diff
@@ -57,6 +58,18 @@ public void onCreated(Item item) {
     }
 
     /**
+     * Called before a job is copied into a new parent, providing the ability to veto the copy operation before it
+     * starts.
+     *
+     * @param src the item being copied
+     * @param parent the proposed parent
+     * @throws Failure to veto the operation.
+     * @since TODO
+     */
+    public void onCheckCopy(Item src, ItemGroup parent) throws Failure {
```

Comment:
There is another reason to use `Failure` and that is if we used another class we would not be able to pick up usage of the class until the baseline has this new class, whereas with `Failure` we can just implement the method on `ItemListener` and it will work on newer jenkins versions: https://github.com/jenkinsci/branch-api-plugin/pull/95/files#diff-b347159a8a8075d4e700065f8668b804R2244